src/test/java/uk/gov/justice/hmpps/prison/util/OffenderProgramProfileBuilder.kt

```diff
@@ -0,0 +1,47 @@
+package uk.gov.justice.hmpps.prison.util
+
+import org.springframework.data.repository.findByIdOrNull
+import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderProgramProfile
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.AgencyLocationRepository
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.CourseActivityRepository
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.OffenderBookingRepository
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.OffenderProgramProfileRepository
+import uk.gov.justice.hmpps.prison.service.BadRequestException
+import java.time.LocalDate
+
+class OffenderProgramProfileBuilder(
+  var prisonId: String = "MDI",
+  var startDate: LocalDate = LocalDate.of(2016, 11, 9),
+  var programStatus: String = "ALLOC",
+  var courseActivityId: Long = -1,
+  var offenderBookingId: Long = -1,
+  var programId: Long = -1
+) : WebClientEntityBuilder() {
+
+  fun save(
+    courseActivityRepository: CourseActivityRepository? = null,
```

Comment:
Maybe a future change will be to add a service method to add a program that is only used by tests; that means we only need the service and not all the repositories at the test level.

Something to think about